src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php

```diff
@@ -294,16 +294,25 @@ public function walkSelectorSource(QOM\SelectorInterface $source)
      *
      * @throws NotImplementedException
```

Comment:
this comment can be updated :-) please also add an explanation what the $root parameter means.